src/privatekey.cpp

```diff
@@ -68,6 +68,14 @@ PrivateKey PrivateKey::FromBytes(const uint8_t* bytes, bool modOrder) {
     return k;
 }
 
+PrivateKey PrivateKey::FromBN(bn_t sk) {
+    PrivateKey k;
+    k.AllocateKeyData();
+    bn_copy(*k.keydata, sk);
+    bn_free(sk);
```

Comment:
We shouldn't free the bn since we didn't create it here, we should free it outside.

Reply:
done